kairosdb/src/main/scala/akka/stream/alpakka/kairosdb/KairosDBSinkStage.scala

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.kairosdb
+
+import akka.Done
+import akka.stream.stage._
+import akka.stream.{Attributes, Inlet, SinkShape}
+import org.kairosdb.client.Client
+import org.kairosdb.client.builder.MetricBuilder
+import org.kairosdb.client.response.Response
+
+import scala.concurrent._
+import scala.util.{Failure, Success}
+
+final case class KairosSinkSettings(parallelism: Int) {
+  require(parallelism > 0)
+
+}
+
+object KairosSinkSettings {
+  val Defaults = KairosSinkSettings(1)
+}
+
+private object NullExecutionContext extends ExecutionContext {
+  override def execute(runnable: Runnable): Unit = ???
+
+  override def reportFailure(cause: Throwable): Unit = ???
+}
+
+class KairosDBSinkStage(settings: KairosSinkSettings,
+                        kairosClient: Client)(implicit executionContext: ExecutionContext = NullExecutionContext)
+    extends GraphStageWithMaterializedValue[SinkShape[MetricBuilder], Future[Done]] {
+
+  val in: Inlet[MetricBuilder] = Inlet("KairosSink.in")
+
+  override val shape: SinkShape[MetricBuilder] = SinkShape(in)
+
+  override def createLogicAndMaterializedValue(inheritedAttributes: Attributes): (GraphStageLogic, Future[Done]) = {
+    val promise = Promise[Done]()
+    val logic = new KairosSinkStageLogic(in, shape, kairosClient, promise, settings)(executionContext)
+
+    (logic, promise.future)
+  }
+}
+
+private[kairosdb] class KairosSinkStageLogic(
+    in: Inlet[MetricBuilder],
+    shape: SinkShape[MetricBuilder],
+    kairosClient: Client,
+    promise: Promise[Done],
+    settings: KairosSinkSettings
+)(executionContext: ExecutionContext)
+    extends GraphStageLogic(shape)
+    with StageLogging {
+
```

Comment:
Actually in the first implementation I did it in this way. 
But later on decided to keep it similar to what already is there in alpakka. e.g. IranMQ implementation
https://github.com/akka/alpakka/blob/master/ironmq/src/main/scala/akka/stream/alpakka/ironmq/IronMqPullStage.scala#L48
But still is still more flexible as client can pass different EC as well.